theano/tensor/opt.py

```diff
@@ -1457,8 +1457,10 @@ def apply(self, fgraph):
 
 # Register it after merge1 optimization at 0. We don't want to track
 # the shape of merged node.
+# As local_shape_to_shape_i is not in fast_compile,
+# dont register this opt in fast_compile.
 theano.compile.mode.optdb.register('ShapeOpt', ShapeOptimizer(),
-                                   0.1, 'fast_run', 'fast_compile')
+                                   0.1, 'fast_run')
```

Comment:
Follow that pattern if you find other opt that fail with the same error (shape_feature isn't there).